grails-app/services/org/pih/warehouse/tableroapi/IndicatorDataService.groovy

```diff
@@ -11,20 +11,47 @@ import org.pih.warehouse.tablero.IndicatorDatasets
 
 class IndicatorDataService {
 
+    def dashboardService
     Date today = new Date()
 
-    DataGraph getExpirationSummaryData(def expirationData) {
-        List listData = []
-        for(item in expirationData){
-            def tmp = item.value? item.value : 0
-            listData.push(tmp)
-        }
+    DataGraph getExpirationSummaryData(Location location, def params) {
+        Integer querySize = params.querySize ? params.querySize.toInteger() - 1 : 5
+        today.clearTime()
+
+        List expirationSummary = []
+        List listLabels = []
+        List expirationAlerts = dashboardService.getExpirationAlerts(location)
         
+        Date tempDate = today.clone()
+
+        expirationAlerts.each {
+            Integer daysCounter = 0
+            if(it.inventoryItem.expires != "never") {
+                if (it.daysToExpiry > 0) {
+                    listLabels = []
+                    for (int i=0; i<=querySize; i++) {
+                        tempDate.set(month: today.month + i, date: i ? 1 : today.date)
```

Comment:
I think you should consider using joda time here, with the `LocalDate` you don't need to reset the time and you will have the `plusMonths()` and `withMonthOfYear()` methods.
To get the name of the moth you can use the `DateTimeFormat.forPattern("MMM").withLocale(Locale.US)` to format the date which will give you the first 3 letters of a month.
To get the max number of days in current month you can use `date.dayOfMonth().getMaximumValue()`.
@jmiranda what do you think?

Reply:
@pmuchowski I'm was just confused in general about what a lot of the date data code is trying to accomplish so I've requested comments on everything, so I could recommend ways to improve it. But as a general rule, I agree, TimeCategory and JodaTime should be considered friends.

Reply:
And in general, err on the side of making the code readable.